src/index.js

```diff
@@ -1,4 +1,151 @@
+// /* eslint-disable no-unused-expressions */
+// expect(someTrueValue).to.be.true;
+
 import './styles/main.scss';
-import CreatToDo from './modules/domDisplay.js';
+import showNotification from './modules/notification';
+
+const form = document.getElementById('todoform');
+const todoInput = document.getElementById('newtodo');
+const todosListEl = document.getElementById('todos-list');
+const refresh = document.getElementById('clear');
+
+// VARS
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let EditTodoId = -1;
+
+// clearAll todo lists when refresh the page
+refresh.addEventListener('click', () => {
+  window.location.reload();
+});
+
+export default refresh;
+// 1st render
+// renderTodos();
+// EDIT A TODO
+const editTodo = (todoId) => {
+  todoInput.value = todos[todoId].value;
+  EditTodoId = todoId;
+};
+
+// RENDER TODOS
+const renderTodos = () => {
+  if (todos.length === 0) {
+    todosListEl.innerHTML = '<center>Nothing to do!</center>';
+    return;
+  }
+  // CLEAR ELEMENT BEFORE A RE-RENDER
+  todosListEl.innerHTML = '';
+  // RENDER TODOS
+  todos.forEach((todo, index) => {
+    todosListEl.innerHTML += `
+      <div class="todo" id=${index}>
+        <i 
+          class="bi ${todo.checked ? 'bi bi-check2-square' : 'bi bi-app'}"
+          style="color : ${todo.color}"
+          data-action="check"
+        ></i>
+        <p class="${todo.checked ? 'checked' : ''}" data-action="check">${todo.value}</p>
+        <i class="bi bi-pencil-square" data-action="edit"></i>
+        <i class="bi bi-trash" data-action="delete"></i>
+      </div>
+      `;
+  });
+};
+
+// CHECK A TODO
+const checkTodo = (todoId) => {
+  todos = todos.map((todo, index) => ({
+    ...todo,
+    checked: index === todoId ? !todo.checked : todo.checked,
+  }));
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+// SAVE TODO
+const saveTodo = () => {
+  const todoValue = todoInput.value;
+  // check if the todo is empty
+  const isEmpty = todoValue === '';
+
+  // check for duplicate todos
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+
+  if (isEmpty) {
+    showNotification("Todo's input is empty");
+  } else if (isDuplicate) {
+    showNotification('Todo already exists!');
+  } else {
+    if (EditTodoId >= 0) {
+      todos = todos.map((todo, index) => ({
+        ...todo,
+        value: index === EditTodoId ? todoValue : todo.value,
+      }));
+      EditTodoId = -1;
+    } else {
+      todos.push(
+        {
+          value: todoValue,
+          checked: false,
+          color: 'orange',
+        },
```

Comment:
- [ ] All the tasks should have the property index set to the value of the new array length (i.e. if a 4th task is added to the list, the index of that task should equal 4) but for you, there is no property of index 🙃 🙃, please kindly fix that.

This is the look of your data in the local storage:
<img width="623" alt="Screen Shot 2022-08-12 at 13 59 31" src="https://user-images.githubusercontent.com/74501523/184351429-6e88512a-5102-4a4e-9da3-92300b942ae9.png">

- [ ] You suppose to have a property of index and not of color 😁 😁